### PR TITLE
Update release notes draft to version v1.30.0-rc.0

### PR DESCRIPTION
**Pull request by satyampsoni**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR updates the Release Notes Draft to k/k v1.30.0-rc.0

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
This is an automated PR generated from `krel The Kubernetes Release Toolbox`

### CODE REVIEW

**Review by fykaa — COMMENTED** [review]
Review added. Thanks @satyampsoni !

File: releases/release-1.30/release-notes/release-notes-draft.json [hunk 1]
```diff
@@ -2170,6 +2220,30 @@
     "duplicate": true,
     "duplicate_kind": true
   },
+  "122692": {
```
Comment (rashansmith) [inline releases/release-1.30/release-notes/release-notes-draft.json:2223]: This should be removed

File: releases/release-1.30/release-notes/release-notes-draft.json [hunk 2]
```diff
@@ -4487,6 +4548,8 @@
     "duplicate": true,
     "duplicate_kind": true
   },
+ 
+
```
Comment (rashansmith) [inline releases/release-1.30/release-notes/release-notes-draft.json:4552]: If these were added manually, try not to change the spacing.

File: releases/release-1.30/release-notes/maps/pr-122946-map.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+pr: 122946
+releasenote:
+  text: Users can now traverse all the pods that are in the scheduler and waiting
+    in the permit stage through method `IterateOverWaitingPods`. In other words,  all
+    waitingPods in scheduler can be obtained from any profiles. Before this commit,
+    each profile could only obtain waitingPods within that profile.
```
[the remark this answers is not in the document]
Comment (ramrodo) [reply releases/release-1.30/release-notes/maps/pr-122946-map.yaml:6]: @satyampsoni could you apply this suggestion of putting backticks to `waitingPods`, please?

**Pull request merged**